src/pantry.js

```diff
@@ -1,97 +1,46 @@
 class Pantry {
   constructor(userIngredients) {
     this.contents = userIngredients;
-    this.filtered
-    this.goodAmount;
-    this.notEnough = []
   }
-
-  determineIfUserHasEnoughIngredients(recipe) {
-    let AllIngredientInPantryIds = this.contents.map(ing =>{
-      return ing.ingredient 
+  getIdsOfPantryIngredients() {
+    return this.contents.map(ing => {
+      return ing.ingredient
     })
-
-     this.filtered = recipe.ingredients.filter(ingredient => {
-      return AllIngredientInPantryIds.includes(ingredient.id)
+  }
+  findRecipeIngredientsInPantry(recipe) {
+    let currentPantryIds = this.getIdsOfPantryIngredients()
+    return recipe.ingredients.filter(ingredient => {
+      return currentPantryIds.includes(ingredient.id)
     })
-
-     this.goodAmount =  this.filtered.filter(ing =>{
-      if (AllIngredientInPantryIds.indexOf(ing.id) > -1) {
-        if (!this.contents[AllIngredientInPantryIds.indexOf(ing.id)].amount >= ing.quantity.amount) {
-          this.notEnough.push(ing)
-        }
-        return this.contents[AllIngredientInPantryIds.indexOf(ing.id)].amount >= ing.quantity.amount
-      }
+  }
+  doesUserHaveRequiredIngredients(recipe) {
+    return this.findRecipeIngredientsInPantry(recipe).length === recipe.ingredients.length
+  }
+  findWhichIngredientsAreShort(recipe) {
+    let notEnough = this.findRecipeIngredientsInPantry(recipe).filter(ing => {
+      let ingredientStock = this.findIngredientInPantry(ing).amount
+      return ingredientStock < ing.quantity.amount
     })
-    return this.goodAmount.length === this.filtered.length
+    return notEnough
   }
-
   findAmountMissing(recipe) {
-//invoke determineIfUser... ()recipe
-    //if returns false => 
-    //find which ingredients are not satisfied by pantry
-  if (!this.determineIfUserHasEnoughIngredients(recipe)) { 
-    return this.notEnough.length
-    }
+    let missingIngredients = this.findWhichIngredientsAreShort(recipe)
+   return missingIngredients.reduce((finalArray, ingredient) => {
+      finalArray.push({ name: ingredient.name, facts: { id: ingredient.id, amountMissing: ingredient.quantity.amount - this.findIngredientInPantry(ingredient).amount}})
```

Comment:
I think line 29 could be refactored so that the objects' key-value pairs exist on multiple lines?